sycl/test/extensions/properties/properties_kernel_named_sub_group_size.cpp

```diff
@@ -0,0 +1,42 @@
+// RUN: %clangxx -fsycl-device-only -S -Xclang -emit-llvm %s -o - | FileCheck %s
+// RUN: %clangxx -fsycl -fsyntax-only -Xclang -verify %s
+// expected-no-diagnostics
+#include <sycl/sycl.hpp>
+
+struct SGSizePrimaryKernelFunctor {
+  SGSizePrimaryKernelFunctor() {}
+
+  void operator()(sycl::nd_item<1>) const {}
+
+  auto get(sycl::ext::oneapi::experimental::properties_tag) const {
+    return sycl::ext::oneapi::experimental::properties{
+        sycl::ext::oneapi::experimental::sub_group_size_primary};
+  }
+};
+
+struct SGSizeAutoKernelFunctor {
+  SGSizeAutoKernelFunctor() {}
+
+  void operator()(sycl::nd_item<1>) const {}
+
+  auto get(sycl::ext::oneapi::experimental::properties_tag) const {
+    return sycl::ext::oneapi::experimental::properties{
+        sycl::ext::oneapi::experimental::sub_group_size_automatic};
+  }
+};
+
+int main() {
+  sycl::queue Q;
+  sycl::nd_range<1> NDRange{6, 2};
+
+  // CHECK: spir_kernel void @{{.*}}SGSizePrimaryKernelFunctor()
+  // CHECK-SAME: !intel_reqd_sub_group_size ![[SGSizeAttr:[0-9]+]]
+  Q.parallel_for(NDRange, SGSizePrimaryKernelFunctor{});
```

Comment:
Do we need to add a test here that the kernel actually executes with the primary sub-group size, as returned by `info::device::primary_sub_group_size`?  Or is that covered by another test somewhere else?

Reply:
Ah, it looks like this device info descriptor is still missing. I will add it. As an aside, should it be renamed to `ext_oneapi_primary_sub_group_size`?